libraries/chain/transaction.cpp

```diff
@@ -163,9 +159,12 @@ vector<transaction_extensions> transaction::validate_and_extract_extensions()con
                   "Transaction extensions are not in the correct order (ascending id types required)"
       );
 
-      results.emplace_back();
+      auto iter = results.emplace(std::piecewise_construct,
+         std::forward_as_tuple(id),
+         std::forward_as_tuple()
+      );
 
-      auto match = decompose_t::extract<transaction_extensions_t>( id, e.second, results.back() );
+      auto match = decompose_t::extract<eosio::chain::transaction_extensions>( id, e.second, iter->second );
```

Comment:
I am not a fan of the 3x different "standards" for this typename:

`eosio::chain::transaction_extensions` , presumably fully-qualified because it collides with the member `transaction::transaction_extensions`

`block_header_extensions`, which isn't fully qualified because its associated member is `block_header::header_extensions`

`block_extension` (not plural), which isn't fully qualified because its associated member is `signed_block::block_extensions` (plural)

Is there a good way to create unity across these?  I suppose as the member names are API-relevant we'd be talking about unifying the typenames